src/shogun/lib/SGVector.cpp

```diff
@@ -129,42 +130,37 @@ SGVector<T>::SGVector(EigenRowVectorXt& vec)
 template <class T>
 SGVector<T>::operator EigenVectorXtMap() const
 {
-	if (on_gpu())
-		SG_SERROR("EigenRowVectorXtMap method not available for vectors on GPU.\n")
+	on_gpu_check();
 	return EigenVectorXtMap(vector, vlen);
 }
 
 template <class T>
 SGVector<T>::operator EigenRowVectorXtMap() const
 {
-	if (on_gpu())
-		SG_SERROR("EigenRowVectorXtMap method not available for vectors on GPU.\n")
+	on_gpu_check();
 	return EigenRowVectorXtMap(vector, vlen);
 }
 
 template<class T>
 void SGVector<T>::zero()
```

Comment:
we are still doing zeroing... as @karlnapf said this interface should slowly die.